#6

VyrostkoOlga opened the pull request:
#6

fix: small design fixes
add: CocoaPods and library for network
fix: colors' scheme of main screen
add: action sheet with reordering, sending to server and loading from server buttons
add: logic for loading from server and sending data to server
add: pushing new notes to server and merging notes from server
add: auth for users
add: loading and pushing notes with userName

Commit:
#6
fix: small design fixes
add: CocoaPods and library for network
fix: colors' scheme of main screen
add: action sheet with reordering, sending to server and loading from server buttons
add: logic for loading from server and sending data to server
add: pushing new notes to server and merging notes from server
add: auth for users
add: loading and pushing notes with userName

Commit:
#6
add: Pods forder to gitignore

mrdekk commented on NNotes/NotesTableViewController.m:
```diff
@@ -78,7 +137,108 @@ -(void) startEditing {
+
+-(void) sendToServer {
+    NSURLSessionConfiguration * configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    AFURLSessionManager * manager = [[AFURLSessionManager alloc] initWithSessionConfiguration: configuration];
```
Менеджер не надо каждый раз создавать

mrdekk commented on NNotes/NotesTableViewController.m:
```diff
@@ -78,7 +137,108 @@ -(void) startEditing {
+
+-(void) loadFromServer {
+    NSURLSessionConfiguration * configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    AFURLSessionManager * manager = [[AFURLSessionManager alloc] initWithSessionConfiguration: configuration];
```
Аналогично

mrdekk commented on NNotes/NotesTableViewController.m:
```diff
@@ -78,7 +137,108 @@ -(void) startEditing {
+    self.reorderModeButton.title = @"✎";
+}
+
+-(void) sendToServer {
```
sendToServer, loadFromServer необходимо вынести в отдельную сущность, пусть это будет NotesService

Commit:
#6
 remove: pods

Commit:
#6
change: functionality of loading from server and pushing to server was moved into separate NotesService

Commit:
#6
fix: creation of session manager only once

mrdekk commented on NNotes/NotesService.m:
```diff
@@ -0,0 +1,112 @@
+
+-(void) sendNotesToServer: (NSArray *) notes {
+    NSURLSessionConfiguration * configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    AFURLSessionManager * manager = [[AFURLSessionManager alloc] initWithSessionConfiguration: configuration];
```
manager надо проинициализировать для сервиса один раз и его переиспользовать в вызовах

VyrostkoOlga replied:
Это сделала, просто в самом последнем коммите. Который "fix: creation of session manager only once"

mrdekk commented on NNotes/NotesService.m:
```diff
@@ -0,0 +1,119 @@
+    NSURLRequest * request = [[AFJSONRequestSerializer serializer] requestWithMethod: @"GET" URLString: [NotesService serverUrl] parameters: [[NSDictionary alloc] initWithObjects: @[[NSString stringWithFormat: @"{\"userName\":\"%@\"}", [[NSUserDefaults standardUserDefaults] objectForKey: @"userName"] ]] forKeys: @[@"query"]] error: nil];
+    NSURLSessionDataTask * dataTask = [self.manager dataTaskWithRequest:request completionHandler:^(NSURLResponse *response, id responseObject, NSError *error) {
+        if (error) {
+            NSLog(@"Error while getting notes");
```
Сообщать пользователю об ошибке

Commit:
#6
add: notification for user when error is occurred when loading notes

mrdekk commented:
OK
